src/data_collector/polygon_data/client.py

```
@@ -238,9 +379,90 @@ def _fetch_paginated_data(self, endpoint: str, params: Optional[Dict] = None) ->
+                    k: v[0] if isinstance(v, list) and len(v) == 1 else v
+                    for k, v in next_params.items()
+                }
+
+                # Make request to next page
+                response: Dict[str, Any] = self._make_request(parsed_url.path, request_params)
+            else:
+                # First page
+                response = self._make_request(endpoint, params)
+
+            # Extract results
+            if "results" in response and response["results"]:
```

Ruff (0.14.3)

Unnecessary key check before dictionary access

Replace with `dict.get`

(RUF019)